src/common/oclengine.cpp

```diff
@@ -125,81 +152,122 @@ void OCLEngine::InitOCL()
-        all_device_contexts[i]->calls[OCL_API_CINC] = cl::Kernel(program, "cinc");
-        all_device_contexts[i]->calls[OCL_API_DEC] = cl::Kernel(program, "dec");
-        all_device_contexts[i]->calls[OCL_API_CDEC] = cl::Kernel(program, "cdec");
-        all_device_contexts[i]->calls[OCL_API_INCC] = cl::Kernel(program, "incc");
-        all_device_contexts[i]->calls[OCL_API_DECC] = cl::Kernel(program, "decc");
-        all_device_contexts[i]->calls[OCL_API_INCS] = cl::Kernel(program, "incs");
-        all_device_contexts[i]->calls[OCL_API_DECS] = cl::Kernel(program, "decs");
-        all_device_contexts[i]->calls[OCL_API_INCSC_1] = cl::Kernel(program, "incsc1");
-        all_device_contexts[i]->calls[OCL_API_DECSC_1] = cl::Kernel(program, "decsc1");
-        all_device_contexts[i]->calls[OCL_API_INCSC_2] = cl::Kernel(program, "incsc2");
-        all_device_contexts[i]->calls[OCL_API_DECSC_2] = cl::Kernel(program, "decsc2");
-        all_device_contexts[i]->calls[OCL_API_INCBCD] = cl::Kernel(program, "incbcd");
-        all_device_contexts[i]->calls[OCL_API_DECBCD] = cl::Kernel(program, "decbcd");
-        all_device_contexts[i]->calls[OCL_API_INCBCDC] = cl::Kernel(program, "incbcdc");
-        all_device_contexts[i]->calls[OCL_API_DECBCDC] = cl::Kernel(program, "decbcdc");
-        all_device_contexts[i]->calls[OCL_API_INDEXEDLDA] = cl::Kernel(program, "indexedLda");
-        all_device_contexts[i]->calls[OCL_API_INDEXEDADC] = cl::Kernel(program, "indexedAdc");
-        all_device_contexts[i]->calls[OCL_API_INDEXEDSBC] = cl::Kernel(program, "indexedSbc");
-        all_device_contexts[i]->calls[OCL_API_APPROXCOMPARE] = cl::Kernel(program, "approxcompare");
-        all_device_contexts[i]->calls[OCL_API_NORMALIZE] = cl::Kernel(program, "nrmlze");
-        all_device_contexts[i]->calls[OCL_API_UPDATENORM] = cl::Kernel(program, "updatenorm");
-        all_device_contexts[i]->calls[OCL_API_APPLYM] = cl::Kernel(program, "applym");
-        all_device_contexts[i]->calls[OCL_API_APPLYMREG] = cl::Kernel(program, "applymreg");
-        all_device_contexts[i]->calls[OCL_API_PHASEFLIP] = cl::Kernel(program, "phaseflip");
-        all_device_contexts[i]->calls[OCL_API_ZEROPHASEFLIP] = cl::Kernel(program, "zerophaseflip");
-        all_device_contexts[i]->calls[OCL_API_CPHASEFLIPIFLESS] = cl::Kernel(program, "cphaseflipifless");
-        all_device_contexts[i]->calls[OCL_API_PHASEFLIPIFLESS] = cl::Kernel(program, "phaseflipifless");
-        all_device_contexts[i]->calls[OCL_API_MUL] = cl::Kernel(program, "mul");
-        all_device_contexts[i]->calls[OCL_API_DIV] = cl::Kernel(program, "div");
-        all_device_contexts[i]->calls[OCL_API_CMUL] = cl::Kernel(program, "cmul");
-        all_device_contexts[i]->calls[OCL_API_CDIV] = cl::Kernel(program, "cdiv");
+        toRet.all_device_contexts.push_back(devCntxt);
+
+        toRet.all_device_contexts[i]->calls[OCL_API_APPLY2X2] = cl::Kernel(program, "apply2x2");
```

Comment:
Create a utility macro for these lines, so that it’s:
`#define OCL_CREATE_CALL(NAME, STR) ...` This will clean up this code substantially.  Or put them into an array and iterate over the array, so the code is more readable.

Reply:
I made the macro. I started pulling the pairs into a vector, but then I realized a couple of drawbacks:

1. The vector initialization is about same # of loc as this code block.
2. This requires that we allocate for the kernel name strings, and keep this allocation around forever.

However, I have created the macro.

Reply:
I pulled it into a vector, but let me know whether I should keep or remove the macro, please.